src/TurboMqtt/Protocol/Mqtt311Decoder.cs

```diff
@@ -417,15 +426,19 @@ internal static bool TryGetPacketLength(ref ReadOnlySpan<byte> span, out int bod
         
         bodyLength = 0;
 
+        var encoded = new byte[5];
         byte encodedByte;
         do
         {
             if (byteCount >= span.Length) return false;
 
-            encodedByte = span[byteCount++];
+            encodedByte = span[byteCount];
+            encoded[byteCount] = encodedByte;
+            byteCount++;
+            
             value += (encodedByte & 0x7F) * multiplier;
             if (multiplier > 128 * 128 * 128) // 128 * 128 * 128
-                return false;
+                throw new Exception($"Malformed remaining length. Value = {value}, encoded: {BitConverter.ToString(encoded)}");
```

Comment:
Debug code to print out the offending bytes if we fail to decode the 7-bit encoded length

Reply:
Also stops the decoder from waiting for future proper length value if we fail